src/fastener.js

```diff
@@ -159,6 +167,9 @@ export function last(z) {const u = up(z); return u && downLast(u)}
 
 export const toZipper = focus => ({focus})
 
+const zipperPathU = (ks, focus) => downPathU(ks, toZipper(focus)) || toZipper()
```

Comment:
Note that `toZipper()`, without an argument, is currently equivalent to `toZipper(undefined)` and is technically not valid as `undefined` does not pass as a valid JSON value.

I'm not 100% sure of what the intention behind the `|| toZipper()` part is.  Perhaps the idea is to make a kind of conditional move: move down path it is valid or stay if not.  Such conditional operations can be expressed using `queryMove` and also just `||`:

```js
downPath(p, z) || z
```

TBH, I'm not sure `zipperPath` should be added.  It seems a bit of a special case and asymmetric.